src/optimize/create_ui_exports_module.js

```diff
@@ -0,0 +1,32 @@
+import dedent from 'dedent';
+
+// We normalize all path separators to `/` in generated files
+function normalizePath(path) {
+  return path.replace(/[\\\/]+/g, '/');
+}
+
+export default function () {
+  if (module.id.indexOf('?') === -1) {
+    throw new Error('create_ui_exports_module loaded without JSON args in module.id');
+  }
+
+  const { type, modules } = JSON.parse(module.id.slice(module.id.indexOf('?') + 1));
+
+  const requires = modules
+    .sort((a, b) => a.localeCompare(b))
+    .map(m => {
+      const req = normalizePath(m);
+      return `{ req: '${req}', module: require('${req}') },`;
```

Comment:
__nit:__ would be great if you can add a comment to clarify why we use this format exactly (`{ req, module}`) and where it's consumed.

Reply:
That format was beneficial for the approach I alluded to in the description, where uiExport modules were being pulled in by the registries rather than being pulled in to decorate the registries... No need for it now.